tests/src/core/testqgsgeonodeconnection.cpp

```diff
@@ -0,0 +1,164 @@
+ ***************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ ***************************************************************************/
+#include "qgstest.h"
+#include <QtTest/QtTest>
+
+#include "qgsgeonodeconnection.h"
+#include "qgssettings.h"
+#include "qgsgeonoderequest.h"
+#include "qgslogger.h"
+#include "qgsmessagelog.h"
+
+/** \ingroup UnitTests
+ * This is a unit test for the QgsGeoConnection class.
+ */
+
+class TestQgsGeoNodeConnection: public QObject
+{
+    Q_OBJECT
+
+  private slots:
+    // will be called before the first testfunction is executed.
+    void initTestCase();
+    // will be called after the last testfunction was executed.
+    void cleanupTestCase()
+    {
+      QgsGeoNodeConnectionUtils::deleteConnection( mGeoNodeConnectionName );
+      QgsGeoNodeConnectionUtils::deleteConnection( mDemoGeoNodeName );
+      QgsGeoNodeConnectionUtils::deleteConnection( mKartozaGeoNodeQGISServerName );
+    }
+    // will be called before each testfunction is executed.
+    void init() {}
+    // will be called after every testfunction.
+    void cleanup() {}
+
+    // Check if we can create geonode connection from database.
+    void testCreation();
+
+    // Test Layer API
+    void testLayerAPI();
+
+    // Test Style API
+    void testStyleAPI();
+
+  private:
+    QString mGeoNodeConnectionName;
+    QString mGeoNodeConnectionURL;
+
+    QString mDemoGeoNodeName;
+    QString mDemoGeoNodeURL;
+
+    QString mKartozaGeoNodeQGISServerName;
+    QString mKartozaGeoNodeQGISServerURL;
+
+    QString mKartozaGeoNodeGeoServerName;
+    QString mKartozaGeoNodeGeoServerURL;
+};
+
+// Runs before all unit tests
+void TestQgsGeoNodeConnection::initTestCase()
+{
+  std::cout << "CTEST_FULL_OUTPUT" << std::endl;
+  mGeoNodeConnectionName = QStringLiteral( "ThisIsAGeoNodeConnection" );
+  mGeoNodeConnectionURL = QStringLiteral( "http://www.thisisageonodeurl.com" );
+  mDemoGeoNodeName = QStringLiteral( "Demo GeoNode" );
+  mDemoGeoNodeURL = QStringLiteral( "http://demo.geonode.org" );
+  mKartozaGeoNodeQGISServerName = QStringLiteral( "Staging Kartoza GeoNode QGIS Server" );
+  mKartozaGeoNodeQGISServerURL = QStringLiteral( "http://staging.geonode.kartoza.com" );
+  mKartozaGeoNodeGeoServerName = QStringLiteral( "Staging Kartoza GeoNode GeoServer" );
+  mKartozaGeoNodeGeoServerURL = QStringLiteral( "http://staginggs.geonode.kartoza.com" );
+
+  // Add Demo GeoNode Connection
+  QgsSettings settings;
+
+  // Testing real server, demo.geonode.org. Need to be changed later.
+  settings.setValue( QgsGeoNodeConnectionUtils::pathGeoNodeConnection() + QStringLiteral( "/%1/url" ).arg( mDemoGeoNodeName ), mDemoGeoNodeURL, QgsSettings::Providers );
```

Comment:
I'd prefer if we could avoid pinging the network here: I think that all tests should be able to run on disconnected environments (not only for latency but for consistancy and reliability), why don't you follow the same route that WFS tests do? I.e. store the response in the test data folder and fake the connection to load responses from disk? @nyalldawson @m-kuhn do we have any good practice here?

Reply:
Thank you so much for you're review @elpaso. Yes I think having mocked responses would be nicer.

Reply:
@myarjunar Can you take care of doing this second request - I will get @nyalldawson to do the first.  Thanks!